macrosynergy/signal/target_positions.py

```diff
@@ -123,14 +239,17 @@ def target_positions(df: pd.DataFrame, cids: List[str], xcat_sig: str, ctypes: L
     :param <List[str]> ctypes: contract types that are traded across markets. They should
         correspond to return categories in the dataframe if the `ret` argument is
         appended. Examples are 'FX' or 'EQ'.
-    :param <Dict[Any, Dict]> baskets: dictionary of basket dictionaries. The inner
-        dictionary takes a string of form <cross_section>_<contract_type> as key and a
-        list of string of the same form as value. The key labels the basket. The value
-        defines the contracts that are used for forming the basket. Pe default the
-        contract have equal weights. An example would be:
-        {{'APC_FX' : ['AUD_FX', ''NZD_FX', 'JPY_FX']},
-         {'APC_EQ' : ['AUD_EQ', ''CNY_EQ', 'INR_EQ', 'JPY_EQ']}}
-        # Todo: has yet to be implemented
+    :param <dict> baskets: a dictionary containing the name of each basket and the
```

Comment:
Every basket needs a unique identifier of weights. Say we have a global FX basket. The weights of individual currencies would have an xcat of the form ctype+basket+"WGT"., such as BRL_FX_GLB_FX_WGT. This appropriate labelling must be done by the Basket class and appropriately named weights returned by the .return_weights() method. Then all we need is to pass on the basket name ("GLB_FX") to targget_positions. The function will track down all the weights, check if they sum up to 1 and use them in the appropriate places.